api/analyzers/source_analyzer.py

```diff
@@ -159,22 +159,17 @@ def second_pass(self, graph: Graph, path: Path) -> None:
                             elif key == "parameters":
                                 graph.connect_entities("PARAMETERS", entity.id, symbol.id)
 
-    def analyze_file(self, file_path: Path, path: Path, graph: Graph) -> None:
-        ext = file_path.suffix
-        logging.info(f"analyze_file: path: {file_path}")
-        logging.info(f"analyze_file: ext: {ext}")
-        if ext not in analyzers:
-            return
-
-        self.first_pass(file_path, [], graph)
-        self.second_pass(graph, path)
+    def analyze_files(self, files: list[Path], path: Path, graph: Graph) -> None:
+        self.first_pass(path, files, [], graph)
+        self.second_pass(graph, files, path)
```

Comment:
I believe we need to wait for all files to complete `first_pass` before running `second_pass`.
Please make sure the code is written in such away that it will be easy to process files in parallel.